### PR TITLE
chore: [M3-6080] - Add a ProductInformationBanner to all product landing pages

### PR DESCRIPTION
**Pull request by mjac0bs**
## Description 📝
We decided to preemptively add this component to our other product landing pages so that when there is an issue with a product, we don't have to hotfix a banner in.

TODO:
- [x] Add banners to create/detail pages and confirm persistent dismissal across landing/create/detail.
- [x] Add banners to empty state landing pages.
- [x] Fix the tiny banner on Kube Create.
- [x] Default to`important warning` styling for all the banners, but allow us to pass in params and modify the product information banners via LD.

> [!IMPORTANT]  
> The changes I ended up making to fulfill the last TODO touch **_many_** files. We could opt not to do this (and just revert back to [1f1aee7](https://github.com/linode/manager/pull/9523/commits/1f1aee7098c4229d60daf7bf3c0a5600ffdf4d3f)), or I'm open to suggestions on a different approach. I was attempting to improve `Notice`, whose props included an excessive collection of booleans (`info`, `marketing`, `error`, `success`, and `warning`) to determine the color of the Notice's border. A border can only be one of those colors, so we shouldn't allow more than one of those props to potentially exist on a Notice at a time.

## Major Changes 🔄
- Products that were previously missing the ProductInformationBanner on their landing, create, detail, and empty state pages now have it: Betas, Domains, Firewalls, Images, Linodes, LoadBalancers, Managed, NodeBalancers, StackScripts, Volumes
- Replaces a number of boolean props on `Notice` with a `variant` defined by the `NoticeVariant` type
- Updates the interface for the Launch Darkly `ProductInformationBannerFlag` to expect a `decoration` object with `important` and `variant` key-value pairs -- all LD variations must have the same type, so these are required and can't be optionally included AFAIK

## Preview 📷
Screenshot 2023-08-09 at 11 33 40 AM

## How to test 🧪
1. **How to set up environment?**
   - I added a new variation (`test`) to the Product Information Banner feature flag in Launch Darkly, then *in the Development environment* changed the Targeting 'default rule' to serve that variation. The flag in Staging and Production environments remains on, but the default rule is set to serve `Empty`, so no test banners show up.
   - Turn on the AGLB and VPC feature flags, as you'll need them to see those  landing pages. 
2. **How to verify changes?**
   - Check out this branch and go to the following urls and verify that you see the banner on all our landing pages:
         - http://localhost:3000/linodes 
         - http://localhost:3000/volumes 
         - http://localhost:3000/loadbalancers 
         - http://localhost:3000/nodebalancers 
         - http://localhost:3000/vpc 
         - http://localhost:3000/firewalls 
         - http://localhost:3000/stackscripts
         - http://localhost:3000/images 
         - http://localhost:3000/domains
         - http://localhost:3000/databases (previously existed)
         - http://localhost:3000/kubernetes/clusters (previously existed)
         - http://localhost:3000/object-storage/buckets (previously existed)
         - http://localhost:3000/longview 
         - http://localhost:3000/linodes/create?type=One-Click 
         - http://localhost:3000/account 
         - http://localhost:3000/betas 
   - Click through some create and details pages and verify the banners are there too. Delete a banner and verify it stays deleted on landing/detail/create pages for that banner key.
   -  Modify the **test** variation of the Launch Darkly feature flag in the **Development** environment and change a banner's variation to some variation type other than `warning` and set "important" to "false. Verify in Cloud that the changed banner updates to the different variation and does not have the important icon.
   - Take a look at the diff and verify that Notices and DismissibleBanners still have the same variant types now as they did when they were receiving the variants directly as props. 
   - In particular, verify that the handful of places where `variant` is assigned to a variable that is not just a string literal, check that that variable assignment is still producing the Notice of the expected variant.
   - Verify that adding banners to all these pages doesn't adversely impact the performance of the app.
3. **How to test?**
- Verify no regressions in e2es or unit tests.

### CODE REVIEW

File: packages/manager/src/features/Domains/DomainsLanding.tsx
Comment (mjac0bs) [inline packages/manager/src/features/Domains/DomainsLanding.tsx:1]: Most of the changes in this file are fixed linting warnings.

**Comment by bnussman-akamai** [comment]
I'm seeing a tiny banner on `/kubernetes/create` 🥺

Screenshot 2023-08-10 at 10 24 57 AM

**Comment by bnussman-akamai** [comment]
Not caused by this PR, but I'm seeing a tiny banner on `/kubernetes/create` 🥺
<details>
  <summary>Screenshot 📷</summary>
  
Screenshot 2023-08-10 at 10 27 27 AM

</details>

**Review by bnussman-akamai — COMMENTED** [review]
Thank you so so so much for fixing the Notice component. This has been long overdue! I

Everything looks good, but I wonder if we could reduce some duplicated `<ProductInformationBanner />`s but putting just one in the file that handles the routing. Would this be a good idea?

Screenshot 2023-08-15 at 9 35 25 AM

**Comment by tyler-akamai** [comment]
- [ ] Turn on the AGLB and VPC feature flags, as you'll need them to see those landing pages.
Verify that you see the banner on all our landing pages:
- [ ] http://localhost:3000/linodes
- [ ] http://localhost:3000/volumes
- [ ] http://localhost:3000/loadbalancers
- [ ] http://localhost:3000/nodebalancers
- [ ] http://localhost:3000/vpc
- [ ] http://localhost:3000/firewalls
- [ ] http://localhost:3000/stackscripts
- [ ] http://localhost:3000/images
- [ ] http://localhost:3000/domains
- [ ] http://localhost:3000/databases (previously existed)
- [ ] http://localhost:3000/kubernetes/clusters (previously existed)
- [ ] http://localhost:3000/object-storage/buckets (previously existed)
- [ ] http://localhost:3000/longview
- [ ] http://localhost:3000/linodes/create?type=One-Click
- [ ] http://localhost:3000/account
- [ ] http://localhost:3000/betas

- [ ] Click through some create and details pages and verify the banners are there too. Delete a banner and verify it stays deleted on landing/detail/create pages for that banner key.
- [ ] Modify the test variation of the Launch Darkly feature flag in the Development environment and change a banner's variation to some variation type other than warning and set "important" to "false. Verify in Cloud that the changed banner updates to the different variation and does not have the important icon.
- [ ] Take a look at the diff and verify that Notices and DismissibleBanners still have the same variant types now as they did when they were receiving the variants directly as props.
In particular, verify that the handful of places where variant is assigned to a variable that is not just a string literal, check that that variable assignment is still producing the Notice of the expected variant.
- [ ] Verify that adding banners to all these pages doesn't adversely impact the performance of the app.
How to test?
- [ ] Verify no regressions in e2es or unit tests.

**Review by tyler-akamai — APPROVED** [review]

**Review by bnussman-akamai — APPROVED** [review]

**Comment by mjac0bs** [comment]
> I wonder if we could reduce some duplicated <ProductInformationBanner />s but putting just one in the file that handles the routing. Would this be a good idea?

@bnussman-akamai - Yep, good call! I made that change in [4143af4](https://github.com/linode/manager/pull/9523/commits/4143af40cffeec2795b46054ad8a9ed918d72891). For posterity, just noting an exception: `<ProductInformationBanner />` isn't included in the Linodes routing file because we have two different banners: one for Linodes Landing and Details and one for Linodes Create (necessary for Marketplace to have its own banner as a product).

**Pull request merged**